data_source.go

```diff
@@ -718,11 +742,20 @@ func (ds *AnySource) PrepareRun(Npresamples int, Nsamples int) error {
 	if ds.nchan <= 0 {
 		return fmt.Errorf("PrepareRun could not run with %d channels (expect > 0)", ds.nchan)
 	}
+
+	// Set up things that specific sources set up only if they have unusual settings.
 	if ds.channelsPerPixel < 1 {
```

Comment:
What case does this handle? This seems like a silent error handling case, maybe we should remove it and be explicit in all cases that it handles?

Reply:
Lines 746-757 handle the fact that many data source concrete classes do not currently
* Set `channelsPerPixel=1`.
* Set channel names and numbering arrays.
* Set up the `ds.groupKeysSorted` map.

So we have two choices:
1. Copy all of this code into all the concrete classes.
2. Put this here, to catch concrete classes that didn't set non-default values.

I feel like it would be nice to have the default values be set up earlier in the process, and merely overwritten by setup of concrete classes. The problem with that is that this default setup can't run until we "sample" the source and learn the number of channels. Anyway, this is my compromise.

Reply:
To be clear: only Lancero needs to set `channelsPerPixel=2`. Only Lancero has "Chan X" and "Err X" channel names. Only Lancero and Abaco have complicated channel groups allowed (the others would have the trivial case: a single from from 1 to N).

Reply:
Actually, I have a better idea. If there's an `AnySource.setUpChannelGroupingNaming` or something, then we could put all of this stuff in it, and the LanceroSource and AbacoSource could override it. I'll do that. This might fit well with the factoring-out-and-testing idea.

Reply:
Just pushed changes that add an `AnySource.PrepareChannels()` and methods of `LanceroSource` and `AbacoSource` that override it. Also added tests for the TDM version of the logic.